### ISSUE TITLE
Provide a stack trace or origin function to onInvokeTask

### ISSUE THREAD

**Issue by ballercat** [ballercat/jest-plugin-must-assert#14]
Allow the user to base whether or not the event should fire via `onInvokeTask`. Currently, it's only done by checking the zone ids. There are edge cases where some libraries should be allowed to still fire async tasks after the test is complete.

This will allow some more advanced tests to work as intended while reducing the number of async warnings. 

### Idea

onInvokeTask could receive something in the form of a function(name) where the task is originating from

`onInvokeTask({ originFunctionName })` 

or the stack trace

`onInvokeTask({ stringifiedStack })`

or both

`onInvokeTask({ stringifiedStack, originFunctionName })`

**Comment by kbruccoleri** [comment]
For enhanced flexibility, could you just pass the entire task to `onInvokeTask`?

**Comment by ballercat** [comment]
I wish I knew how to do that, but from what I recall (and it's been a while) we don't have access to the stack at the point where `onInvokeTask` is called. We throw an error once the task is ignored and then leverage the Zone library to generate a stack for us. Because everything is wrapped in zones, the "raw" stack is pretty difficult to parse as it contains lots of internal zone library functions.

Might be worth taking a second look at it at some point and seeing if we can trick Zone into prettifying a stack before we call `onInvokeTask` though.

**Comment by kbruccoleri** [comment]
Thanks for responding! Very glad to come across this project, I think it's a really powerful tool.

I opened up a PR here that I think would work: #22 

It would be a breaking change though for anyone who manually configured the `onInvokeTask` handler, since I swap out `taskSource`, `taskType` for the `task` that both are derived from.

For context, I found that the `enzyme` library actually creates some unresolved Promises unfortunately when determining if a React component is lazy. 

With this change, I was able to check the `task`'s stack trace to see if was created by that package, and then ignore it, as it was overshadowing any issues with the tests themselves.
